core/components/atoms/chip/__tests__/Chip.test.tsx

```diff
@@ -138,4 +138,74 @@ describe('Chip component', () => {
     );
     expect(queryByTestId('DesignSystem-Chip--GenericChip')).toHaveClass('Chip-selection--selectedDisabled');
   });
+
+  it('renders chip component with custom label', () => {
+    const labelText = 'Chip';
+    const customLabel = <Text>{labelText}</Text>;
+    const { queryByTestId } = render(<Chip label={customLabel} name={labelText} />);
+    expect(queryByTestId('DesignSystem-Chip--GenericChip')).toBeInTheDocument();
+    expect(queryByTestId('DesignSystem-Text')).toHaveTextContent(labelText);
+  });
+
+  it('renders chip component with multiple labels', () => {
+    const labelText1 = 'Chip1';
+    const labelText2 = 'Chip2';
+    const customLabel = (
+      <span>
+        <Text>{labelText1}</Text>
+        <Text>{labelText2}</Text>
+      </span>
+    );
+    const { queryAllByTestId } = render(<Chip label={customLabel} name="Chip" />);
+    const labels = queryAllByTestId('DesignSystem-Text');
+    expect(labels[0]).toHaveTextContent(labelText1);
+    expect(labels[1]).toHaveTextContent(labelText2);
+  });
+
+  it('renders chip component with styled labels', () => {
+    const labelText = 'Chip';
+    const customLabel = <Text color="primary">{labelText}</Text>;
+    const { queryByTestId } = render(<Chip label={customLabel} name={labelText} />);
+    expect(queryByTestId('DesignSystem-Text')).toHaveClass('color-primary');
+  });
+
+  it('renders chip component with onClick event trigger', () => {
+    let text = 'Custom';
+    const { getByTestId } = render(
+      <Chip
+        label="Chip"
+        name="label"
+        type="action"
+        disabled={false}
+        selected={true}
+        onClick={(name: any) => {
+          text = name;
+        }}
```

Comment:
Please pass custom component inside `label` and object inside `name` and then check for arguments passed in `onClick` event.  You can use `expect(onSelectFn.mock.calls[1][0]).toStrictEqual([0, 2]);`  to check the first arg of the second call to the mock function.